src/core/main.js

```diff
@@ -344,6 +344,8 @@ class p5 {
       this._events.devicemotion = null;
     }
 
+    this.registerMethod('init', this._updateWindowSize);
+
     // Function to invoke registered hooks before or after events such as preload, setup, and pre/post draw.
```

Comment:
Oh actually, maybe one thing worth considering: if we only register the method here, then I believe any addons that attach init handlers earlier will end up running before this, so their handlers won't have `windowWidth` and `windowHeight` yet. That's maybe not the end of the world, but we can probably ensure this gets run first if instead of registering a handler, we just call `_updateWindowSize()` explicitly right before running the other init handlers (which happens near the end of the constructor)?